enterprise/backend/src/metabase_enterprise/scim/auth.clj

```diff
@@ -15,12 +15,18 @@
        (u.password/verify-password api-key "" expected-api-key)
        (mw.session/do-useless-hash)))))
 
+(def ^:private error-schema-uri "urn:ietf:params:scim:api:messages:2.0:Error")
+
 (defn +scim-auth
-  "Middleware that returns a 401 response if `request` does not have a valid SCIM API key"
+  "Middleware that returns a 401 response if `request` does not have a valid SCIM API key, and/or if
+  SCIM is not enabled."
   [handler]
   (fn [request respond raise]
     (let [authorization-header (get-in request [:headers "authorization"])
           [_ api-key]          (re-matches #"Bearer (.*)" (or authorization-header ""))]
-      (if (validate-scim-api-key api-key)
+      (if (and (scim/scim-enabled) (validate-scim-api-key api-key))
         (handler request respond raise)
-        (respond req.util/response-unauthentic)))))
+        (respond {:status 401
+                  :body   {:schemas [error-schema-uri]
+                           :status  401
```

Comment:
This seems redundant but Okta's test suite was looking for both an error code in the request itself and in the request body 🤷